code/modules/mob/living/simple_animal/abnormality/he/der_freischutz.dm

```diff
@@ -23,6 +35,122 @@
 	gift_type =  /datum/ego_gifts/magicbullet
 	abnormality_origin = ABNORMALITY_ORIGIN_LOBOTOMY
 
+	var/can_act = TRUE
+	var/bullet_cooldown
+	var/bullet_cooldown_time = 7 SECONDS
+	var/bullet_fire_delay = 1.5 SECONDS
+	var/bullet_max_range = 50
+
+	//PLAYABLES ATTACKS (action in this case)
+	attack_action_types = list(/datum/action/innate/abnormality_attack/toggle/der_freischutz_zoom)
+
+/datum/action/innate/abnormality_attack/toggle/der_freischutz_zoom
+	name = "Toggle Sniper Sight"
+	button_icon_state = "zoom_toggle0"
+	chosen_message = span_warning("You activate your sniper sight.")
+	button_icon_toggle_activated = "zoom_toggle1"
+	toggle_message = span_warning("You deactivate your sniper sight.")
+	button_icon_toggle_deactivated = "zoom_toggle0"
+	var/zoom_out_amt = 5.5
+	var/zoom_amt = 10
+	var/original_sight
+
+/datum/action/innate/abnormality_attack/toggle/der_freischutz_zoom/New(Target)
+	..()
+	original_sight = owner.sight
+
+/datum/action/innate/abnormality_attack/toggle/der_freischutz_zoom/Activate()
+	ActivateSignals()
+	owner.sight |= SEE_TURFS | SEE_MOBS | SEE_THRU
+	owner.regenerate_icons()
+	owner.client.view_size.zoomOut(zoom_out_amt, zoom_amt, owner.dir)
+	. = ..()
+
+/datum/action/innate/abnormality_attack/toggle/der_freischutz_zoom/proc/ActivateSignals()
+	SIGNAL_HANDLER
+
+	RegisterSignal(owner, COMSIG_MOVABLE_MOVED, PROC_REF(Deactivate))
+	RegisterSignal(owner, COMSIG_ATOM_DIR_CHANGE, PROC_REF(Rotate))
+
+/datum/action/innate/abnormality_attack/toggle/der_freischutz_zoom/Deactivate()
+	DeactivateSignals()
+	owner.sight = original_sight
+	owner.regenerate_icons()
+	owner.client.view_size.zoomIn()
+	. = ..()
```

Comment:
When `. = ..()` is at the end of a proc, it should be `return ..()` instead
```suggestion
	return ..()
```

Reply:
Adressed in newest commit.